gcloud/datastore/client.py

```diff
@@ -51,53 +130,189 @@ def __init__(self, dataset_id=None, namespace=None, connection=None):
         self.namespace = namespace
 
     def get(self, key, missing=None, deferred=None):
-        """Proxy to :func:`gcloud.datastore.api.get`.
+        """Retrieve an entity from a single key (if it exists).
 
-        Passes our ``dataset_id``.
+        .. note::
+
+           This is just a thin wrapper over :method:`get_multi`.
+           The backend API does not make a distinction between a single key or
+           multiple keys in a lookup request.
+
+        :type key: :class:`gcloud.datastore.key.Key`
+        :param key: The key to be retrieved from the datastore.
+
+        :type missing: an empty list or None.
+        :param missing: If a list is passed, the key-only entities returned
+                        by the backend as "missing" will be copied into it.
+                        Use only as a keyword param.
+
+        :type deferred: an empty list or None.
+        :param deferred: If a list is passed, the keys returned
+                         by the backend as "deferred" will be copied into it.
+                         Use only as a keyword param.
+
+        :rtype: :class:`gcloud.datastore.entity.Entity` or ``NoneType``
+        :returns: The requested entity if it exists.
         """
-        return get(key, missing=missing, deferred=deferred,
-                   connection=self.connection, dataset_id=self.dataset_id)
+        entities = self.get_multi(keys=[key], missing=missing,
+                                  deferred=deferred)
+        if entities:
+            return entities[0]
 
     def get_multi(self, keys, missing=None, deferred=None):
-        """Proxy to :func:`gcloud.datastore.api.get_multi`.
+        """Retrieve entities, along with their attributes.
 
-        Passes our ``dataset_id``.
+        :type keys: list of :class:`gcloud.datastore.key.Key`
+        :param keys: The keys to be retrieved from the datastore.
+
+        :type missing: an empty list or None.
+        :param missing: If a list is passed, the key-only entities returned
+                        by the backend as "missing" will be copied into it.
+                        Use only as a keyword param.
+
+        :type deferred: an empty list or None.
+        :param deferred: If a list is passed, the keys returned
+                         by the backend as "deferred" will be copied into it.
+                         Use only as a keyword param.
+
+        :rtype: list of :class:`gcloud.datastore.entity.Entity`
+        :returns: The requested entities.
+        :raises: ValueError if one or more of ``keys`` has a dataset ID which
+                 does not match our dataset ID.
         """
-        return get_multi(keys, missing=missing, deferred=deferred,
-                         connection=self.connection,
-                         dataset_id=self.dataset_id)
+        if not keys:
+            return []
+
+        ids = list(set([key.dataset_id for key in keys]))
```

Comment:
I just realized the cast to list is not needed, but somewhat irrelevant to this PR.